### PR TITLE
HITL - Add remote mouse input support.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changeset adds remote mouse input.
* Mouse buttons and scrolling are supported.
* Mouse position is still not supported (mouse ray).

## How Has This Been Tested

Tested locally and remotely with Unity client.

## Types of changes

- **\[Development\]**

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-hitl/habitat_hitl/_internal/gui_application.py [hunk 1]
```diff
@@ -48,30 +49,30 @@ def key_press_event(self, event: Application.KeyEvent) -> None:
...
                     wrapper._key_held.remove(key)
                 wrapper._key_up.add(key)
 
     def mouse_press_event(self, event: Application.MouseEvent) -> None:
-        mouse_button = event.button
-        GuiInput.validate_mouse_button(mouse_button)
-        for wrapper in self._gui_inputs:
-            wrapper._mouse_button_held.add(mouse_button)
-            wrapper._mouse_button_down.add(mouse_button)
+        key = MagnumKeyConverter.convert_mouse_key(event.button)
```
Comment (jturner65) [inline habitat-hitl/habitat_hitl/_internal/gui_application.py:60]: Maybe still call it mouse button?  Kind of weird to see a 'key' in a mouse event handler. Not a big deal though

File: habitat-hitl/habitat_hitl/core/remote_client_state.py [hunk 2]
```diff
@@ -156,12 +156,11 @@ def _update_input_state(self, client_states):
...
-            # TODO: Add mouse support
-            # mouse_json = (
-            #    client_state["mouse"] if "mouse" in client_state else None
-            # )
+            mouse_json = (
+                client_state["mouse"] if "mouse" in client_state else None
+            )
 
-            if input_json is not None:
+            if input_json != None:
```
Comment (aclegg3) [inline habitat-hitl/habitat_hitl/core/remote_client_state.py:163]: I think "is not" is the preferred pythonic way to check None still

**Review by aclegg3 — APPROVED** [review]
Looks good. Minor nit about None checking (multiple occurrences)

File: habitat-hitl/habitat_hitl/core/remote_client_state.py (hunk 2 above)
Comment (0mdc) [reply habitat-hitl/habitat_hitl/core/remote_client_state.py:163]: That's right!

File: habitat-hitl/habitat_hitl/_internal/gui_application.py (hunk 1 above)
Comment (0mdc) [reply habitat-hitl/habitat_hitl/_internal/gui_application.py:60]: This makes more sense yeah.

**Pull request merged**